pkg/controller/elasticsearch/stackmon/config.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stackmon
+
+import (
+	_ "embed" // for the beats config files
+	"fmt"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/label"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/user"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Environments variables and paths to the Elasticsearch CA certificates used in the beats configuration to describe
+// how to connect to Elasticsearch.
+// Warning: environment variables and CA cert paths defined below are also used in the embedded files.
+var (
+	esSourceURLEnvVarKey      = "ES_SOURCE_URL"
+	esSourceURLEnvVarValue    = "https://localhost:9200"
```

Comment:
what if https is disabled?

Reply:
I forgot that. I will see to support it. It shouldn't be difficult.

Reply:
This is covered by the latest version using go templating.